src/jvm/main/org/jetbrains/kotlinx/lincheck/util/AtomicMethods.kt

```diff
@@ -122,8 +123,92 @@ internal fun isAtomicFieldUpdaterClass(className: String) =
+    "java.lang.invoke.VarHandleBytes${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleBytes${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleBytes${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleBytes${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleChars${'$'}Array",
+    "java.lang.invoke.VarHandleChars${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleChars${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleChars${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleChars${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleDoubles${'$'}Array",
+    "java.lang.invoke.VarHandleDoubles${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleDoubles${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleDoubles${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleDoubles${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleFloats${'$'}Array",
+    "java.lang.invoke.VarHandleFloats${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleFloats${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleFloats${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleFloats${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleInts${'$'}Array",
+    "java.lang.invoke.VarHandleInts${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleInts${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleInts${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleInts${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleLongs${'$'}Array",
+    "java.lang.invoke.VarHandleLongs${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleLongs${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleLongs${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleLongs${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleReferences${'$'}Array",
+    "java.lang.invoke.VarHandleReferences${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleReferences${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleReferences${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleReferences${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleObjects${'$'}Array",
+    "java.lang.invoke.VarHandleObjects${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleObjects${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleObjects${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleObjects${'$'}FieldStaticReadWrite",
+    "java.lang.invoke.VarHandleShorts${'$'}Array",
+    "java.lang.invoke.VarHandleShorts${'$'}FieldInstanceReadOnly",
+    "java.lang.invoke.VarHandleShorts${'$'}FieldInstanceReadWrite",
+    "java.lang.invoke.VarHandleShorts${'$'}FieldStaticReadOnly",
+    "java.lang.invoke.VarHandleShorts${'$'}FieldStaticReadWrite",
+)
+
+@OptIn(ExperimentalContracts::class)
+internal fun isVarHandle(obj: Any?): Boolean {
+    contract { returns(true) implies (obj != null) }
```

Comment:
Can we try something similar to the code below? 
```
    VAR_HANDLE_CLASS?.isAssignableFrom(obj)
```

Reply:
BTW, if the concern is the matching against a hard-coded set of strings of var-handle class names, then we already have a similar code here:
https://github.com/JetBrains/lincheck/blob/800f590ef453c3eab5c87b7264c29e580060cec4/src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/VarHandleNames.kt#L26

Reply:
I also agree + I benchmarked and found that it is slower than mine.